### PR TITLE
Bug 1462205 - Delete binding shouldn't show up when all bindings are pending deletion

### PR DESCRIPTION
**Pull request by jwforres**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1462205

### CODE REVIEW

File: app/scripts/controllers/overview.js [hunk 1]
```diff
@@ -1101,6 +1101,7 @@ function OverviewController($scope,
     // - API objects by binding name
     state.bindingsByApplicationUID = {};
     state.applicationsByBinding = {};
+    state.validBindingsByApplicationUID = {};
```
Comment (jwforres) [inline app/scripts/controllers/overview.js]: im not in love with the variable names here or in service-row, open to other suggestions

**Comment by jwforres** [comment]
This also fixes the text display to say Deleting with the spinner instead of Pending

File: app/scripts/controllers/overview.js (hunk 1 above)
Comment (spadgett) [inline app/scripts/controllers/overview.js]: I don't have a great suggestion :(

 `valid` sort of implies ready to me... `deleteableBindingsByApplicationUID` too on the nose?

Comment (jwforres) [inline app/scripts/controllers/overview.js]: yeah i started to call it undeletedBindingsByApplicationUID, its just... weird

Comment (spadgett) [inline app/scripts/controllers/overview.js]: This is specifically for the delete binding action, so I don't mind a name that makes that clear even if it doesn't roll off the tongue.

Alternatively we could put the filter in overview-list-row and call it `deleteableBindings`.

**Comment by jwforres** [comment]
@spadgett updated the variable names, I also changed it to only pass the deleteable bindings into the unbind service directive

**Comment by spadgett** [comment]
@benjaminapetersen FYI, probably conflicts with #1742

**Comment by spadgett** [comment]
[merge][severity:blocker]

**Pull request merged**